### PR TITLE
Limits the number of class-combinations to avoid freeze

### PR DESCRIPTION
**Pull request by vwochnik**
This is especially useful for the modernizr sites which have dozens of classes attached to their `<html>` element.

### CODE REVIEW

**Comment by vwochnik** [comment]
Please merge this. It will fix the freeze when there are many classes on one element.

File: src/getCombinations.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+function kCombinations( result, items, data, start, end, index, k )
```
Comment (AvraamMavridis) [inline src/getCombinations.js]: missing jsdoc block

File: src/getCombinations.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        data[index] = items[i];
+        kCombinations( result, items, data, i + 1, end, index + 1, k );
+    }
+}
+
+/**
+ * Returns all the possible selector combinations
+ */
+export function getCombinations( items, k )
```
Comment (AvraamMavridis) [inline src/getCombinations.js]: pls add jsdoc block (`@param`, `@returns`)

File: test/unique-selector.js [hunk 3]
```diff
@@ -35,6 +35,15 @@ describe( 'Unique Selector Tests', () =>
     expect( uniqueSelector ).to.equal( 'body > :nth-child(1)' );
   } );
 
+  it( 'Classes', () =>
```
Comment (AvraamMavridis) [inline test/unique-selector.js]: this should explain what is testing.

File: src/getCombinations.js (hunk 1 above)
Comment (vwochnik) [inline src/getCombinations.js]: Ok, will create one.

File: test/unique-selector.js (hunk 3 above)
Comment (vwochnik) [inline test/unique-selector.js]: Well, I named the tests after your tests which are not any more descriptive than this one either.

Comment (AvraamMavridis) [inline test/unique-selector.js]: Right, I see, there are not more descriptive indeed. Sorry, I cant remember the codebase, I havent work on the project the last months.

**Pull request merged**